### DISCUSSION TITLE
how to handle Disaster recovery in SW spec

### DISCUSSION THREAD

**Discussion by yzhao244** [serverlessworkflow/specification#610] [Q&A] [answered]
Hello, I got a use case from a cloud customer regarding how to handle DR requirements in our specification.
Basically, the customer has an action for doing document transcoding. For example, Function4 is used as the main transcoding service. However, the customer wants to provide the DR capability. If the Function4 is suspended, the service fails to be invoked. Workflow can automatically invoke the other services.

My understanding would be that the customer expects the action contains weight factor. Each time the service with a higher weight is invoked first, if succeed, then go to next step, If the operation fails to be invoked, the service with a lower weight is retried.

However, this is only my thought when looking into this DR use case. :) .. Please let me know any ideas how to handle this use case.

![image](https://user-images.githubusercontent.com/15642771/160688919-fabd825d-5406-4f3e-9446-ec78473081b5.png)

**Comment by tsurdilo** [comment]
I'm not sure if fully understand the requirement but to me the mentioned "weights" are just the order of actions you put in the 
operation state "actions" array. Ideally you would just put them in this "weight" order in this array and we would have ability catch error, and say "continue"..but we don't have that atm as onError has to come to either transition or end definition atm.
Can you open issue for this please?

You always have the option to chain these actions one per operation state and in onError for the raised error transition to the next operation state with next action that needs to be performed. little painful but would work. 
Idk what DR pattern/requirements are but also seems to be kinda similar to maybe compensation? If it is then we do support compensation as well. Let me know

**Comment by ricardozanini** [comment]
Are there any patterns or specifications for DR handling that you know, @yzhao244? Maybe we could take a deep look into that to see how we can accommodate this pattern here or make it easier to implement.

**Comment by cdavernas** [comment, marked as answer]
You can now achieve that using the `try` task defined in the `1.x.x` version of the DSL. As for defining weights, you can do so out of the box using a `do` task - that defines a task to set the weight - combined to a `switch` task that determines the invocation to perform based on the defined weights.

**Reply by yzhao244** [reply]
Cool, thx for the input @cdavernas
